scipy/optimize/minpack.py

```diff
@@ -671,6 +672,20 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
     >>> plt.show()
 
     """
+
+    def check_len(z):
```

Comment:
This whole function isn't correct, sorry. Can you please remove it.

Reply:
At first I was getting some errors in checks when I added the test function. Thats why I also added cases for inf but that was not the actual reason for that. I will jest remove those parts and will keep the main logic for checking size.

Reply:
I have removed the `inf` checks. I have kept size checks and also checks if someone passes a single int or float value

Reply:
> also checks if someone passes a single int or float value

You can't do type checks like that. All relevant inf/nan checks and conversion to ndarray are already taken care of in the code:
```
    if check_finite:
        ydata = np.asarray_chkfinite(ydata)
    else:
        ydata = np.asarray(ydata)

    if isinstance(xdata, (list, tuple, np.ndarray)):
        # `xdata` is passed straight to the user-defined `f`, so allow
        # non-array_like `xdata`.
        if check_finite:
            xdata = np.asarray_chkfinite(xdata)
        else:
            xdata = np.asarray(xdata)
```
so if you want to check for a scalar input, just `xdata.ndim == 0` is enough.

and really, please remove the `check_len` function, this is a simple enough check that it can be done in a single if-statement inline

Reply:
Ohk, so I will remove the function and will replace it with the if-statement line, that would be enough?

Reply:
indeed

Reply:
Also I am skipping the scaler part coz while using the `ndim` I am getting error that `AttributeError: 'int' object has no attribute 'ndim'`